Dirty parent when removng a child node.

Ms2ger opened the pull request:
<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/8315)

<!-- Reviewable:end -->

Commit:
Dirty parent when removng a child node.

Ms2ger commented:
@bors-servo r=pcwalton

Rebase of #8139

bors-servo commented:
:pushpin: Commit e024138 has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton e0241384cd1fefbaaa6f210085000d03b4b74bbf -->

bors-servo commented:
:hourglass: Testing commit e024138 with merge 56576b9...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/194)

eefriedman commented:
```
Unexpected Results
==================

/css21_dev/html4/block-in-inline-remove-000.htm
-----------------------------------------------
PASS expected FAIL [Parent]
/css21_dev/html4/block-in-inline-remove-002.htm
-----------------------------------------------
PASS expected FAIL [Parent]
```

On both Mac and Linux.

Commit:
Updated test expectations.

The WPT CSS tests block-in-inline-remove-000.htm and block-in-inline-remove-002.htm.ini now pass.

frewsxcv commented:
@bors-servo r=pcwalton

bors-servo commented:
:pushpin: Commit 43997b4 has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton 43997b472490feef8f70486de4f7b266a4bc0f14 -->

bors-servo commented:
:hourglass: Testing commit 43997b4 with merge 02198ce...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/225)

eefriedman commented:
@bors-servo retry #8311.

bors-servo commented:
:hourglass: Testing commit 43997b4 with merge ec10bc4...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/237)

jdm commented:
On one hand, I don't recall seeing these failures before:
```

  ▶ FAIL [expected PASS] /_mozilla/css/nth_last_of_type_pseudo_a.html
  └   → /_mozilla/css/nth_last_of_type_pseudo_a.html 4a7970049c8e886dceb8f5a5dfc3479c0b1fa0be
/_mozilla/css/nth_last_of_type_pseudo_b.html f3c5e7f8004718c306b56f0183c3c0f233daf37d
Testing 4a7970049c8e886dceb8f5a5dfc3479c0b1fa0be == f3c5e7f8004718c306b56f0183c3c0f233daf37d

  ▶ TIMEOUT [expected PASS] /_mozilla/css/stacked_layers.html
```
On the other hand I don't think they showed up on previous runs :<

asajeffrey commented:
@bors-servo retry

asajeffrey commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 43997b4 with merge acb2819...

asajeffrey commented:
I ran those failing tests locally, and they passed :( intermittent failures, oh joy.

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/252)

jdm commented:
@bors-servo: r=pcwalton

bors-servo commented:
:pushpin: Commit 43997b4 has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton 43997b472490feef8f70486de4f7b266a4bc0f14 -->

jdm commented:
@bors-servo: retry

bors-servo commented:
:hourglass: Trying commit 43997b4 with merge 79df584...

jdm commented:
@asajeffrey I think you need to rebase this again so we can try to merge it.

jdm commented:
@bors-servo: r=pcwalton

bors-servo commented:
:pushpin: Commit 2fb5d6d has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton 2fb5d6d18b65b02803fdc4c080dc012e6de0d20e -->

asajeffrey commented:
Rebased, and took the opportunity to fix a typo in the commit message.

eefriedman commented:
@bors-servo: r=pcwalton

bors-servo commented:
:pushpin: Commit 558b0b6 has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton 558b0b6c29fb6255676bbc62ee7d8d8c4b8fa5e1 -->

bors-servo commented:
:hourglass: Testing commit 558b0b6 with merge c54558d...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/284), [gonk](http://build.servo.org/builders/gonk/builds/291), [linux-dev](http://build.servo.org/builders/linux-dev/builds/287), [linux-rel](http://build.servo.org/builders/linux-rel/builds/310), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/299), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/295), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/311)

waddlesplash commented on tests/wpt/metadata-css/css21_dev/html4/block-in-inline-remove-000.htm.ini:
```diff
@@ -1,3 +1,3 @@
 [block-in-inline-remove-000.htm]
   type: reftest
-  expected: FAIL
+  expected: PASS
```
Isn't the norm to just delete the file now, and the testdriver will assume a pass is expected?
